File: core/src/de/project/ice/ecs/components/BreathComponent.java
package de.project.ice.ecs.components;

import com.badlogic.gdx.math.Vector2;

/**
 * Component for idle breathing animation of characters
 */
public class BreathComponent implements IceComponent<BreathComponent>
{

    public Vector2 curScale = new Vector2(0f, 0f);
    public Vector2 scaleValue = new Vector2(0f, 0f);
    public float duration = 1f;
    public float time = 0.0f;

    @Override
    public void reset()
    {
        curScale = new Vector2();
        scaleValue = new Vector2();
        duration = 1f;
        time = 0.0f;
    }

    @Override
    public void copyTo(BreathComponent copy)
    {
    }
}

File: core/src/de/project/ice/ecs/components/BreathComponent.kt
package de.project.ice.ecs.components

import de.project.ice.annotations.Property

/**
 * Component for idle breathing animation of characters
 */
class BreathComponent : CopyableIceComponent {
    @Property("The current scaling factor", true)
    val curScale = Vector2(0f, 0f)
    @Property("The maximal scale of a breath")
    val scaleValue = Vector2(0f, 0f)
    @Property("The duration of a breath")
    var duration = 1f
    @Property("The current animation time", true)
    var time = 0.0f

    override fun reset() {
        curScale.set(0f, 0f)
        scaleValue.set(0f, 0f)
        duration = 1f
        time = 0.0f
    }

    override fun copyTo(copy: CopyableIceComponent) {
        if (copy is BreathComponent) {
            copy.curScale.set(curScale)
            copy.scaleValue.set(scaleValue)
            copy.duration = duration
            copy.time = time
        }
    }
}


